tests/docker/docker_test.go

```diff
@@ -0,0 +1,207 @@
+package lim
+
+import (
+	"flag"
+	"fmt"
+	"github.com/docker/docker/pkg/namesgenerator"
+	"github.com/lf-edge/eden/pkg/device"
+	"github.com/lf-edge/eden/pkg/expect"
+	"github.com/lf-edge/eden/pkg/projects"
+	"github.com/lf-edge/eden/pkg/utils"
+	"github.com/lf-edge/eve/api/go/info"
+	log "github.com/sirupsen/logrus"
+	"math/rand"
+	"os"
+	"testing"
+	"time"
+)
+
+// This test deploys the docker://nginx app into EVE with port forwarding 8028->80
+// wait for the RUNNING state and checks access to HTTP endpoint
+// and removes app from EVE
+
+var (
+	timewait     = flag.Int("timewait", 600, "Timewait for items waiting in seconds")
+	tc           *projects.TestContext
+	externalIP   string
+	appName      string
+	externalPort = 8028
+	appLink      = "docker://nginx"
+	portPublish  = []string{fmt.Sprintf("%d:80", externalPort)}
+)
+
+// TestMain is used to provide setup and teardown for the rest of the
+// tests. As part of setup we make sure that context has a slice of
+// EVE instances that we can operate on. For any action, if the instance
+// is not specified explicitly it is assumed to be the first one in the slice
+func TestMain(m *testing.M) {
```

Comment:
Isn't it time we put all of this into a helper method someplace? It seems that this part of TestMain is 100% common to all the tests, no?

Reply:
Part of logic moved to `tc.AddEdgeNodesFromDescription()`